### PR TITLE
scan: Implement see more button in transaction list

### LINKED ISSUE #1557: See more button in transaction list
image

As the image above, the tx list's height is too long. I think we should show 3 messages and make a `See more` button for showing all messages.

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1557 
image
image

### CODE REVIEW

File: scan/src/components/TxMessages.re [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ]);
+  let showContainer = style([display(`flex), marginTop(`px(10))]);
+};
+
+[@react.component]
+let make = (~txHash: Hash.t, ~messages, ~width: int, ~success: bool, ~rawLog: string) => {
+  let (overflowed, setOverflowed) = React.useState(_ => false);
+  let (expanded, setExpanded) = React.useState(_ => false);
+
+  // ReactDOMRe._getElementById("messageWrapper") |> Belt.Option.getExn;
```
Comment (prin-r) [inline scan/src/components/TxMessages.re]: please remove unused comment

File: scan/src/components/TxMessages.re [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  let showButton =
+    style([
+      display(`flex),
+      backgroundColor(Colors.gray3),
+      borderRadius(`px(30)),
+      width(`px(60)),
+      alignItems(`center),
+      justifyContent(`center),
+      fontSize(`px(10)),
+      alignItems(`center),
```
Comment (evilpeach) [inline scan/src/components/TxMessages.re]: duplicated

File: scan/src/components/TxMessages.re [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+           <React.Fragment key={(txHash |> Hash.toHex) ++ (i |> string_of_int)}>
+             <VSpacing size=Spacing.sm />
+             <Msg msg width />
+             <VSpacing size=Spacing.sm />
+           </React.Fragment>
+         )
+       ->React.array}
+      {success
+         ? React.null
+         : <div> <Text value={"Error: " ++ rawLog} code=true size=Text.Sm breakAll=true /> </div>}
```
Comment (evilpeach) [inline scan/src/components/TxMessages.re]: Remove unused `<div>`

File: scan/src/components/TxMessages.re [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  let divID = "messageWrapper" ++ (txHash |> Hash.toHex);
+
+  React.useEffect0(_ => {
+    let x = ReactDOMRe._getElementById(divID) |> Belt.Option.getExn;
+    let divHeight = ReactDOMRe.domElementToObj(x)##clientHeight;
+    divHeight > 60 ? setOverflowed(_ => true) : ();
+    None;
+  });
+  <>
+    <div id=divID className={Styles.msgContainer(overflowed)}>
```
Comment (prin-r) [inline scan/src/components/TxMessages.re]: please try useRef
https://reactjs.org/docs/hooks-reference.html#useref

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by prin-r — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**